frontend/www/js/omegaup/course/students.ts

```diff
@@ -11,6 +13,28 @@ OmegaUp.on('ready', function () {
     components: {
       'omegaup-course-viewprogress': course_ViewProgress,
     },
+    data: () => ({
+      completeStudentsProgress: [] as types.StudentProgressInCourse[],
+    }),
+    async mounted() {
+      let nextPage: undefined | number = 1;
+      let completeStudentsProgress: types.StudentProgressInCourse[] = [];
+      while (nextPage) {
+        await api.Course.studentsProgress({
+          page: nextPage,
+          length: 1,
+          course: payload.course.alias,
+        })
+          .then((response) => {
+            completeStudentsProgress = completeStudentsProgress.concat(
+              response.progress,
+            );
+            nextPage = response.nextPage;
+          })
+          .catch(ui.apiError);
```

Comment:
```suggestion
        try {
          const response = await api.Course.studentsProgress({
            page: nextPage,
            length: 1,
            course: payload.course.alias,
          });
          completeStudentsProgress = completeStudentsProgress.concat(
            response.progress,
          );
          nextPage = response.nextPage;
        } catch (e) {
          ui.apiError(e);
          break;
        }
```

cuando se usa `await`, no se debe usar `.then`. también, hay que rodear esta llamada con un `try/catch` en vez de usar el `.catch`.